### ISSUE TITLE
Intellisense references are too aggressively cached

### ISSUE THREAD

**Issue by davidfowl** [dotnet/roslyn-project-system#785]
- Make a new netstandard csproj
- Edit the csproj and add JSON.NET
   ```xml
    <PackageReference Include="Newtonsoft.Json">
      <Version>9.0.1</Version>
    </PackageReference>
   ```
- Change Class1.cs to this:
    ```C#
    using System;

    namespace ClassLibrary10
    {
        public class Class1
        {
            public Class1()
            {
                Newtonsoft.Json.JsonConvert.SerializeObject(null);
            }
        }
    }
    ```
- Observe everything is highlighted properly
- Now remove the package reference node with Newtonsoft.Json

## Expected

Without building, errors in the IDE

## Actual

Everything is dandy

![image](https://cloud.githubusercontent.com/assets/95136/20475275/c98d3b74-af7f-11e6-8aad-7b3f506c3ddc.png)

**Comment by srivatsn** [comment]
Probably related to #784 - the language service is not being told about add\remove of the reference?

**Comment by wjk** [comment]
I'm seeing the converse of what @davidfowl is reporting: I add a `<PackageReference>` to the csproj, run `dotnet restore`, and find that the autocompletion does not present APIs in the added package. If I manually type the full name of an API, it is colored properly, and member and parameter IntelliSense works properly; only completion of the type names themselves seems to be broken. This condition persists after I reload the project, reload the solution, or even quit and restart Visual Studio entirely. Yeah, I feel that the IntelliSense information is being cached too aggressively.

@srivatsn Is there anything I could help out with to fix this? This is driving me quietly insane, given how quickly I iterate on my projects. Thanks!

**Comment by 333fred** [comment]
@davidfowl, I'm not able to repro this issue on the latest internal builds.

@wjk, I'm not seeing that issue either. I'm confused what you mean by manually running `dotnet restore` though; when you modify the project file to add a `PackageReference`, we should be automatically picking up the change and restoring the new packages, without you having to do anything extra. Could you tell me what steps you're performing?

**Comment by srivatsn** [comment]
Closing this since we are not seeing these issues now. Please reactivate if this happens again.
